q2_assembly/bowtie2/utils.py

```diff
@@ -139,7 +139,7 @@ def _get_subdir_from_path(fp: str, input_type: str = "contigs"):
         subdir (str): Subdir to be created, based on the given input.
     """
     if input_type.lower() == "contigs":
-        return os.path.basename(fp).split("_", maxsplit=1)[0]
+        return os.path.basename(fp.replace("_contigs.fa", ""))
```

Comment:
I recommend calling `_get_sample_from_path` here, to centralize the logic of finding the sample id in the file name in one place.